src/Dialect/Mlir/DialectBuilder.cpp

```diff
@@ -627,12 +627,14 @@ Value MathBuilder::constant(Type type, double val) const {
           // If unsigned, create a signless constant, then cast it to unsigned.
           if (elementType.isUnsignedInteger()) {
             Type signlessTy = b().getIntegerType(width);
-            constant = b().create<arith::ConstantOp>(loc(),
-                b().getIntegerAttr(signlessTy, APInt(width, (int64_t)val)));
+            constant = b().create<arith::ConstantOp>(
+                loc(), b().getIntegerAttr(signlessTy,
+                           APInt(width, (int64_t)val, false, true)));
```

Comment:
Please change to c++ style cast to prevent code scan finding.

static_cast<int64_t>(val) instead of (int64_t)val

Also line 637.

Reply:
@mikeessen Thanks for keeping track of the casts. Could you look into seeing if there is away to have the compiler issue a warning (probably clang for us since that what we are using)? I am afraid that I may not be as vigilant as we should be on this issue (speaking for me on the code I modify...).

That would be greatly appreciated. Thanks

Reply:
> @mikeessen Thanks for keeping track of the casts. Could you look into seeing if there is away to have the compiler issue a warning (probably clang for us since that what we are using)? I am afraid that I may not be as vigilant as we should be on this issue (speaking for me on the code I modify...).
> 
> That would be greatly appreciated. Thanks

Thanks, I will take a look.